internal/global/trace.go

```diff
@@ -139,6 +140,30 @@ func (t *tracer) Start(ctx context.Context, name string, opts ...trace.SpanStart
 		return delegate.(trace.Tracer).Start(ctx, name, opts...)
 	}
 
+	return t.newSpan(ctx, autoInstEnabled, name, opts)
+}
+
+// autoInstEnabled determines if the auto-instrumentation SDK span is returned
+// from the tracer when not backed by a delegate and auto-instrumentation has
+// attached to this process.
+//
+// The auto-instrumentation is expected to overwrite this value to true when it
+// attaches. By default, this will point to false and mean a tracer will return
+// a nonRecordingSpan by default.
+var autoInstEnabled = new(bool)
+
+func (t *tracer) newSpan(ctx context.Context, autoSpan *bool, name string, opts []trace.SpanStartOption) (context.Context, trace.Span) {
+	// autoInstEnabled is passed to newSpan via the autoSpan parameter. This is
+	// so the auto-instrumentation can define a uprobe for (*t).newSpan and be
+	// provided with the address of the bool autoInstEnabled points to. It
+	// needs to be a parameter so that pointer can be reliably determined, it
+	// should not be read from the global.
+
+	if *autoSpan {
+		tracer := sdk.GetTracerProvider().Tracer(t.name, t.opts...)
```

Comment:
As long as `sdk.GetTracerProvider()` is here and is not changed then the circular dependency  of Go modules should not be a problem. I think we should make some automation e.g. using golangci-lint to make sure we will limit usage of `auto/sdk` to bare minimum. Also in `auto/sdk` add something to minimize the possibility of changing this function signature.

Reply:
Yeah, I guess the ask is if we can make `auto/sdk` stable.

I'm not opposed to bringing that request back to the auto SIG. But it may mean there is a greater chance of a `v2` for this package if we find an issue implementing this here.

Reply:
Alternativly it may be good enough to request stability of only this one function without making a V1.

Reply:
> Alternativly it may be good enough to request stability of only this one function without making a V1.

I mean, that I can almost guarantee. It's just that it would only be codified in a commented agreement between developers instead of a version number.

Reply:
> I mean, that I can almost guarantee. It's just that it would only be codified in a commented agreement between developers instead of a version number.

It seems good enough to me. 

What do you think of adding a `depguard` deny rule for `auto/sdk`  in `.golangci.yml` and add a `nolint` comment in this file? This should reduce the risk of accidental increase of coupling to `auto/sdk`.

> That is something we could do. But, I'm wondering if we should be embedding a noop there`.

You mean change https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/f9241162a20bd4beb20d4347db3d8a77280920b1/sdk/trace.go#L34

to `type tracerProvider struct{ noop.TracerProvider }`?

If so then, I think this should be done. It should mitigate a chicken-egg problem.

Reply:
> > That is something we could do. But, I'm wondering if we should be embedding a noop there`.
> 
> You mean change https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/f9241162a20bd4beb20d4347db3d8a77280920b1/sdk/trace.go#L34
> 
> to `type tracerProvider struct{ noop.TracerProvider }`?
> 
> If so then, I think it would be good and should mitigate a chicken-egg problem when introducing a new method to Trace API.

Yeah, that's what I'm thinking: :+1: 

Shouldn't be needed for this PR, but would be needed before we release this.

Reply:
Updated depguard. There is no way to ignore use outside of the import statement. I've just ignored these files.

Reply:
> @pellared should we just have the `auto/sdk{,/telemetry}` packages live here?
> 
> We originally published them under `auto` to allow development in the SIG they are designed for, but I wonder if the better user-story is one where they are hosted under `otel/internal/global/...`?
> 
> It would add code for auto-instrumentation here and add development burden, but it would also ensure there is never an `otel` release that could be broke by an independent upgrade of `auto/sdk`.

This would require https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1207 or something similar so we would no longer depend on `auto/sdk/telemetry` in the `auto` module.

Reply:
> We originally published them under `auto` to allow development in the SIG they are designed for, but I wonder if the better user-story is one where they are hosted under `otel/internal/global/...`?

I have a "feeling" that this would be a better design as this repository would be less (not?) dependent on `go.opentelemetry.io/auto`. Maybe it would be good to explore?

PS. This kind of what I meant by

> Maybe auto could be sub-packages of API modules?

Sorry that my comments  are not very precise, but I also also having unprecise thoughts 😅